src/app/modules/workspace/add-update-criteria/add-update-criteria.component.ts

```diff
@@ -9,28 +14,88 @@ import {MatDialog, MatDialogRef, MAT_DIALOG_DATA} from '@angular/material/dialog
   styleUrls: ['./add-update-criteria.component.scss']
 })
 export class AddUpdateCriteriaComponent implements OnInit {
-  dynamicForm: FormGroup;
+  @ViewChild(DynamicFormComponent) form: DynamicFormComponent;
   submitted: boolean = false;
+  criteriaDetails: any;
+  formdata: any;
+  assesmentType: any;
+  frameworkId: any;
+  criteriaID: any;
+  onload = {
+    buttonName: 'ADD CRITERIA',
+    submitClick: false
+  }
   constructor(private formBuilder: FormBuilder,
-    private dialogRef: MatDialogRef<AddUpdateCriteriaComponent>,) { }
+    private communityService: CommunityService,
+    private commonService: CommonService,
+    private route: ActivatedRoute,
+    private dialogRef: MatDialogRef<AddUpdateCriteriaComponent>,
+    @Optional() @Inject(MAT_DIALOG_DATA) public data) {
 
-  ngOnInit() {
+    this.formdata = this.data['fieldsForCriteria'];
+    this.criteriaID = this.formdata?.criteriaID;
+    if(this.formdata.action == 'Add') {
+     this.onload = {
```

Comment:
recheck the logic

Reply:
This is required for the handling both update and add models